app/src/main/java/io/getstream/chat/android/core/poc/library/json/adapters/QuerySortAdapter.kt

```diff
@@ -0,0 +1,21 @@
+package io.getstream.chat.android.core.poc.library.json.adapters
+
+import com.google.gson.Gson
+import com.google.gson.TypeAdapter
+import com.google.gson.stream.JsonReader
+import com.google.gson.stream.JsonWriter
+import io.getstream.chat.android.core.poc.library.requests.QuerySort
+import java.io.IOException
+import java.util.*
+
+class QuerySortAdapter : TypeAdapter<QuerySort>() {
+    override fun write(out: JsonWriter?, value: QuerySort?) {
+        val adapter = Gson().getAdapter(ArrayList::class.java)
```

Comment:
There is no need to create instance of `Gson`. See how `FilterObjectAdapter` reuses `Gson` instance.